client/rpc-spec-v2/src/chain_head/chain_head.rs

```diff
@@ -203,18 +204,26 @@ where
 		let subscriptions = self.subscriptions.clone();
 
 		let fut = async move {
-			let Some(handle) = subscriptions.get_subscription(&follow_subscription) else {
-				// Invalid invalid subscription ID.
-				let _ = sink.send(&ChainHeadEvent::<String>::Disjoint);
-				return
+			let _block_guard = match subscriptions.lock_block(&follow_subscription, hash) {
```

Comment:
Just for clarification: We only guard here to complete the current operation in case the subscription unpins in the meantime (assuming its the last subscription pinning)?

Reply:
Yep, It may be possible, although highly unlikely that the last subscription of the block calls unpin while we perform some operation with that block.